src/SourceBuild/content/eng/detect-binaries.sh

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env bash
+
+### Usage: $0
+###
+###   Prepares and runs the binary tooling to detect binaries in the VMR. Default behavior is to report any binaries
+###   not found in the allowed-binaries file. To remove binaries not specified in the allowed-binaries file, pass --clean.
+###
+### Options:
+###   --clean                    Clean the VMR of binaries not in the specified allowed-binaries file.
+###   --allowed-binaries-file    Path to the file containing the list of binaries to be
+###                              ignored for either cleaning or validating.
+###                              Defaults to eng/allowed-vmr-binaries.txt for validate.
+###                              Defaults to eng/allowed-sb-binaries.txt for clean.
+###   --log-level <level>        Set the log level for the binary tooling. Defaults to Debug.
+###   --with-packages            Use the specified directory as the packages source feed.
+###                              Defaults to online dotnet-public and dotnet-libraries feeds.
+###   --with-sdk                 Use the specified directory as the dotnet SDK.
+###                              Defaults to .dotnet.
+
+set -euo pipefail
+IFS=$'\n\t'
+
+source="${BASH_SOURCE[0]}"
+REPO_ROOT="$( cd -P "$( dirname "$0" )/../" && pwd )"
+BINARY_TOOL="$REPO_ROOT/eng/tools/BinaryToolKit"
+
+function print_help () {
+    sed -n '/^### /,/^$/p' "$source" | cut -b 5-
+}
+
+defaultDotnetSdk="$REPO_ROOT/.dotnet"
+defaultAllowedBinariesFile="$REPO_ROOT/eng/allowed-vmr-binaries.txt"
+
+# Set default values
+allowedBinariesFile=$defaultAllowedBinariesFile
+mode='validate'
+logLevel='Debug'
+propsDir=''
+packagesDir=''
+dotnetSdk=$defaultDotnetSdk
+
+positional_args=()
+while :; do
+  if [ $# -le 0 ]; then
+    break
+  fi
+  lowerI="$(echo "$1" | awk '{print tolower($0)}')"
+  case $lowerI in
+    "-?"|-h|--help)
+      print_help
+      exit 0
+      ;;
+    --clean)
+      mode="clean"
+      if [ "$allowedBinariesFile" == "$defaultAllowedBinariesFile" ]; then
```

Comment:
Why is this logic necessary.  Isn't this covered by https://github.com/dotnet/installer/pull/19043/files#diff-67a5785b6bbe4a34f17d01965c598eaf99b0f3601e1d9a3e4ef714bac47af18eR35 and/or https://github.com/dotnet/installer/pull/19043/files#diff-67a5785b6bbe4a34f17d01965c598eaf99b0f3601e1d9a3e4ef714bac47af18eR56

Reply:
Without this logic, it's possible that a custom allowed-binaries file would be overridden if --allowed-binaries-file is passed before --clean. For example:

`./eng/detect-binaries.sh --allowed-binaries-file temp --clean` -> file is "$REPO_ROOT/eng/allowed-sb-binaries.txt"
`./eng/detect-binaries.sh --clean --allowed-binaries-file temp` -> file is "temp"

Reply:
IC now.  I don't think the default should change depending on the mode.  The script should have one allowed-binaries default.  The SB prep script should be explicitly flowing in the SD allowed binaries when calling here.  Changing defaults depending on other state is confusing and can lead to usage mistakes.

Reply:
Just to clarify the behavior based on your last comment:

The default behavior when `--clean` is passed is to remove any "new" binaries from the VMR. To remove non-sb-allowed binaries, allowed-sb-binaries.txt would have to be passed via --allowed-binary-file. Is this the behavior you're envisioning?